OrbitService/ProcessServiceImpl.cpp

```diff
@@ -40,30 +41,9 @@ Status ProcessServiceImpl::GetModuleList(ServerContext*,
   int32_t pid = request->process_id();
   LOG("Sending modules for process %d", pid);
 
-  absl::MutexLock lock(&mutex_);
-
-  std::shared_ptr<Process> process = process_list_.GetProcess(pid);
-  if (process == nullptr) {
-    return Status(
-        StatusCode::NOT_FOUND,
-        absl::StrFormat("The process with pid=%d was not found", pid));
-  }
-
-  process->ListModules();
-
-  const std::map<uint64_t, std::shared_ptr<Module>>& modules =
-      process->GetModules();
-
-  for (auto& it : modules) {
-    std::shared_ptr<Module> module = it.second;
-    ModuleInfo* module_info = response->add_modules();
-
-    module_info->set_name(module->m_Name);
-    module_info->set_file_path(module->m_FullName);
-    module_info->set_file_size(module->m_PdbSize);
-    module_info->set_address_start(module->m_AddressStart);
-    module_info->set_address_end(module->m_AddressEnd);
-    module_info->set_build_id(module->m_DebugSignature);
+  std::vector<ModuleInfo> module_infos = LinuxUtils::ListModules(pid);
```

Comment:
It seems unlikely that ListModules never fails.. what if the process we are looking at has recently died?

Reply:
I am going to refactor that in the other PR I mentioned in the last comment. `LinuxUtils::ListModules()` needs to be able to fail and since that is part of OrbitCore I will put it together with the other OrbitCore Error related changes.